core-java-modules/core-java-concurrency-collections-2/src/main/java/com/baeldung/concurrent/lock/SingleLock.java

```diff
@@ -0,0 +1,44 @@
+package com.baeldung.concurrent.lock;
+
+import java.util.Map;
+import java.util.concurrent.locks.ReentrantLock;
+
+import com.google.common.base.Supplier;
+
+public class SingleLock extends ConcurrentAccessExperiment {
+    ReentrantLock lock;
+
+    public SingleLock() {
+        lock = new ReentrantLock();
+    }
+
+    protected synchronized Supplier<?> putSupplier(Map<String,String> map, int key) {
```

Comment:
I'm not sure I understand the point of these methods being `synchronized` if they are using programmatic locks. I've always understood that these are alternatives - you can use locks instead of `synchronized` as an optimization.

Reply:
ok, it was added from a previous thread we had. removed now.